pkg/api/pod/util.go

```diff
@@ -652,6 +653,53 @@ func dropDisabledCSIVolumeSourceAlphaFields(podSpec, oldPodSpec *api.PodSpec) {
 	}
 }
 
+// dropDisabledNodeInclusionPolicyFields removes disabled fields from PodSpec related
+// to NodeInclusionPolicy only if it is not used by the old spec.
+func dropDisabledNodeInclusionPolicyFields(podSpec, oldPodSpec *api.PodSpec) {
+	if !utilfeature.DefaultFeatureGate.Enabled(features.PodTopologySpreadNodePolicy) &&
+		!nodeAffinityPolicyInUse(oldPodSpec) &&
+		podSpec != nil {
+		for i := range podSpec.TopologySpreadConstraints {
+			podSpec.TopologySpreadConstraints[i].NodeAffinityPolicy = nil
+		}
+	}
+	if !utilfeature.DefaultFeatureGate.Enabled(features.PodTopologySpreadNodePolicy) &&
+		!nodeTaintsPolicyInUse(oldPodSpec) &&
+		podSpec != nil {
+		for i := range podSpec.TopologySpreadConstraints {
+			podSpec.TopologySpreadConstraints[i].NodeTaintsPolicy = nil
+		}
+	}
```

Comment:
This can be merged to be slighly more performant:

```go
	if !utilfeature.DefaultFeatureGate.Enabled(features.PodTopologySpreadNodePolicy) &&
		podSpec != nil {
		if !nodeAffinityPolicyInUse(oldPodSpec) {
			for ...
		}
		if !nodeTaintsPolicyInUse(oldPodSpec) {
			for ...
		}
	}
```